feat: Convenience APIs for tags

gavindsouza opened the pull request:
Add a new tag on your document by:
```python
>>> frappe.get_doc("Artist", "Arctic Monkeys").add_tag("AM")

```


View list of tags on your document by:
```python
>>> frappe.get_doc("Artist", "Arctic Monkeys").get_tags()
["AM", "rock"]
```

Docs Link: https://github.com/frappe/frappe_docs/pull/17

gavindsouza committed:
feat: Add tags to your document easily

View list of tags on your document by:
    frappe.get_doc(dt, dn).tags

Add a new tag on your document by:
    frappe.get_doc(dt, dn).add_tag(tag)

gavindsouza commented:
Using properties can have conflicts with existing Document fields. Could change `tags` to a `get_tags` function instead

scmmishra commented:
> Using properties can have conflicts with existing Document fields. Could change tags to a get_tags function instead

Yeah, it makes sense!

gavindsouza committed:
fix: change property tags to get_tags function

avoid conflicts/ambiguity referring to tags property of a document
and field names 'tags'

surajshetty3416 committed:
Merge branch 'develop' into i-like-tags

surajshetty3416 committed:
Merge branch 'develop' into i-like-tags

surajshetty3416 committed:
Merge branch 'develop' into i-like-tags

surajshetty3416 committed:
Merge branch 'develop' into i-like-tags
